src/classes/mech/ActiveState.ts

```diff
@@ -1164,24 +1173,36 @@ class ActiveState {
   }
 
   public static Deserialize(pilot: Pilot, data: IActiveStateData): ActiveState {
-    const mech = pilot.Mechs.find(x => x.ID === data.active_mech_id)
-    if (!mech) return new ActiveState(pilot)
     const s = new ActiveState(pilot)
-    s._mech = mech
-    s._stage = (data.stage as Stage) || Stage.Narrative
-    s._round = data.turn || 1
-    s._mission = data.mission || 0
-    s._actions = data.actions || 2
-    s._braced = data.braced
-    s._overcharged = data.overcharged
-    s._prepare = data.prepare
-    s._bracedCooldown = data.bracedCooldown
-    s._redundant = data.redundant
-    s._pilot_mounted = data.mounted
-    s._stats = data.stats ? data.stats : ActiveState.NewCombatStats()
-    s._deployed = data.deployed ? data.deployed.map(x => Deployable.Deserialize(x, s._mech)) : []
+    s.Update(pilot, data)
     return s
   }
+
+  public Update(pilot: Pilot, data: IActiveStateData, sync?: boolean): void {
```

Comment:
A new ActiveState function was needed to update an ActiveState object in place for syncs.